.github/workflows/ci-generate.yaml

```
@@ -0,0 +1,51 @@
+name: CI
+
+on:
+  pull_request:
+     branches:
+       - main
+
+jobs:
+  drift-check:
+    runs-on: ubuntu-latest
+    env:
+      REPO: ${{ github.event.repository.name }}
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Install Rust toolchain
+        uses: dtolnay/rust-toolchain@stable
+
+      - name: Install cargo-generate
+        uses: taiki-e/install-action@v2
+        with:
+          tool: cargo-generate
+
+      - name: Retain a copy of every generated file
+        run: |
+          for template_file in $(find . -type f -name "*.template"); do
+            generated_file="${template_file%.template}"
+            if [ -f "$generated_file" ]; then
+              # Keep a copy to diff with the output of the next step
+              cp "$generated_file" "${generated_file}.check"
```

```suggestion
          for template_file in $(find . -type f -name '*.template'); do
            generated_file="${template_file%.template}"
            if [[ -f "{$generated_file}" ]]; then
              # Keep a copy to diff with the output of the next step.
              cp "${generated_file}" "${generated_file}.check"
```
Style / convention stuff